ql/src/java/org/apache/hadoop/hive/ql/metadata/PartitionIterable.java

```diff
@@ -99,7 +99,11 @@ private void getNextBatch() {
           batchCounter++;
         }
         try {
-          batchIter = db.getPartitionsByNames(table, nameBatch, getColStats).iterator();
+          if (!isAuthRequired) {
```

Comment:
nit: Using if (isAuthRequired) will make this more readable

Reply:
done